torch/_dynamo/compiled_autograd.py

```diff
@@ -368,7 +472,10 @@ def is_impure(node):
                 or (node.op == "call_function" and node.target in _impure_targets)
             )
 
+        before = len(list(self.fx_tracer.graph.nodes))
```

Comment:
This is doing O(n) work to check length, `graph.nodes` has an O(1) length method:
https://github.com/pytorch/pytorch/blob/eaa24821f27385fcee1e8d4c21012c2397b90976/torch/fx/graph.py#L298-L299